### PR TITLE
Add CI to docs + Password protect pages

### PR DESCRIPTION
**Pull request by jawadqur**
Link to JIRA ticket if there is one: 

### Improvements
- Add a gh action to deploy doc site to gh pages w/ password

### CODE REVIEW

**Review by smvgarcia — APPROVED** [review]

File: gen3/mkdocs.yml [hunk 1]
```diff
@@ -122,8 +125,9 @@ plugins:
       blog_toc: true
   - mkdocs-video:
       is_video: True
-extra_css:
-  - stylesheets/extra.css
+  - encryptcontent:
+      password_inventory:
+        _global: 'gen3'
```
Comment (Avantol13) [inline gen3/mkdocs.yml:130]: I realize this isn't necessarily intended to be secure, but having this plain text sorta defeats the purpose of trying to hide the content on the branches. I think this needs to be a real Github secret

**Review by Avantol13 — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+permissions:
+  contents: write
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
```
Comment (Avantol13) [inline .github/workflows/docs-ci.yaml:16]: Do you have to configure this at all?

File: .github/workflows/docs-ci.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: docs-ci
+on:
+  push:
+    branches:
+      - master
```
Comment (Avantol13) [inline .github/workflows/docs-ci.yaml:5]: Doesn't this mean we're only showing this staged information when the PR is merged? Shouldn't this be setup to allow people to visualize their changes on a branch _before_ a merge?

File: .github/workflows/docs-ci.yaml (hunk 2 above)
Comment (jawadqur) [reply .github/workflows/docs-ci.yaml:16]: https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions 

According to this it is needed. I think it's so github actions is able to publish to pages.

File: gen3/mkdocs.yml (hunk 1 above)
Comment (jawadqur) [reply gen3/mkdocs.yml:130]: This is just so that random people don't run into it for now, until we're ready to publish.

Comment (Avantol13) [reply gen3/mkdocs.yml:130]: Yeah but the repo is public so anyone can see this. So long as it's well understood that this isn't actually protecting against anything and folks writing the documentation understand that this does not actually _prevent_ anyone from seeing the content, then it's okay. But configuring a secret on the repo and using that as a var here isn't too complicated and would alleviate my concern entirely

File: .github/workflows/docs-ci.yaml (hunk 3 above)
Comment (Avantol13) [reply .github/workflows/docs-ci.yaml:5]: resolved in Slack, this is not intended to be a final solution for staging/publishing

**Review by smvgarcia — APPROVED** [review]
I think this will work to solve michael's problem

File: gen3/mkdocs.yml (hunk 1 above)
Comment (jawadqur) [reply gen3/mkdocs.yml:130]: Yeah, the intention behind this password is for crawlers not to index this page, not to hide the content from the public.

**Pull request merged**